pkg/po/lex.go

```diff
@@ -90,84 +128,102 @@ func lex(r io.Reader) ([]Token, error) {
-		token = TokenTypeReference
-	case strings.HasPrefix(line, "#| msgid_plural"):
-		token = TokenTypeMsgidPluralPrevUntStrPlural
-	case strings.HasPrefix(line, "#| msgid"):
-		token = TokenTypeMsgidPrevUntStr
-	case strings.HasPrefix(line, "#| msgctxt"):
-		token = TokenTypeMsgctxtPreviousContext
-	case strings.HasPrefix(line, "#,"):
-		token = TokenTypeFlag
-	case strings.HasPrefix(line, "#"):
-		token = TokenTypeTranslatorComment
+		return nil, errors.New("unknown line prefix")
+	case strings.HasPrefix(line, headerLanguagePrefix):
+		tokenType, linePrefix = TokenTypeHeaderLanguage, headerLanguagePrefix
+	case strings.HasPrefix(line, headerPluralFormsPrefix):
+		tokenType, linePrefix = TokenTypeHeaderPluralForms, headerPluralFormsPrefix
+	case strings.HasPrefix(line, headerTranslatorPrefix):
+		tokenType, linePrefix = TokenTypeHeaderTranslator, headerTranslatorPrefix
+	case strings.HasPrefix(line, headerProjectIDVersionPrefix):
+		tokenType, linePrefix = TokenTypeHeaderProjectIDVersion, headerProjectIDVersionPrefix
+	case strings.HasPrefix(line, headerPOTCreationDatePrefix):
+		tokenType, linePrefix = TokenTypeHeaderPOTCreationDate, headerPOTCreationDatePrefix
+	case strings.HasPrefix(line, headerPORevisionDatePrefix):
+		tokenType, linePrefix = TokenTypeHeaderPORevisionDate, headerPORevisionDatePrefix
+	case strings.HasPrefix(line, headerLastTranslatorPrefix):
+		tokenType, linePrefix = TokenTypeHeaderLastTranslator, headerLastTranslatorPrefix
+	case strings.HasPrefix(line, headerLanguageTeamPrefix):
+		tokenType, linePrefix = TokenTypeHeaderLanguageTeam, headerLanguageTeamPrefix
+	case strings.HasPrefix(line, headerMIMEVersionPrefix):
+		tokenType, linePrefix = TokenTypeHeaderMIMEVersion, headerMIMEVersionPrefix
+	case strings.HasPrefix(line, headerContentTypePrefix):
+		tokenType, linePrefix = TokenTypeHeaderContentType, headerContentTypePrefix
+	case strings.HasPrefix(line, headerContentTransferEncodingPrefix):
+		tokenType, linePrefix = TokenTypeHeaderContentTransferEncoding, headerContentTransferEncodingPrefix
+	case strings.HasPrefix(line, headerReportMsgidBugsToPrefix):
+		tokenType, linePrefix = TokenTypeHeaderReportMsgidBugsTo, headerReportMsgidBugsToPrefix
+	case strings.HasPrefix(line, headerXGeneratorPrefix):
+		tokenType, linePrefix = TokenTypeHeaderXGenerator, headerXGeneratorPrefix
+	case strings.HasPrefix(line, headerGeneratedByPrefix):
+		tokenType, linePrefix = TokenTypeHeaderGeneratedBy, headerGeneratedByPrefix
+	case strings.HasPrefix(line, msgCtxtPrefix):
+		tokenType, linePrefix = TokenTypeMsgCtxt, msgCtxtPrefix
+	case strings.HasPrefix(line, pluralMsgIDPrefix):
+		tokenType, linePrefix = TokenTypePluralMsgID, pluralMsgIDPrefix
+	case strings.HasPrefix(line, msgIDPrefix):
+		tokenType, linePrefix = TokenTypeMsgID, msgIDPrefix
+	case strings.HasPrefix(line, pluralMsgStrPrefix):
+		if tokenType, linePrefix = TokenTypePluralMsgStr,
+			regexp.MustCompile(`^msgstr\[(?:\d+|\*)\]`).FindString(line); linePrefix == "" {
```

Comment:
I believe that compiling regex whenever the case is executed is not efficient. 
I would suggest moving this to package level variable